src/modules/README.md

```diff
@@ -1,78 +1,95 @@
 
 To create a module:
 
-## 1. Create a Service
+## 1. Create a Data Model
+
+A data model represents a table in the database. You create a data model in a TypeScript or JavaScript file under the `models` directory of a module.
+
+For example, create the file `src/modules/blog/models/post.ts` with the following content:
+
+```ts
+import { model } from "@medusajs/framework/utils"
+
+const Post = model.define("post", {
+  id: model.id().primaryKey(),
+  title: model.text(),
+})
+
+export default Post
+```
+
+## 2. Create a Service
 
 A module must define a service. A service is a TypeScript or JavaScript class holding methods related to a business logic or commerce functionality.
 
-For example, create the file `src/modules/hello/service.ts` with the following content:
+For example, create the file `src/modules/blog/service.ts` with the following content:
 
-```ts title="src/modules/hello/service.ts"
-export default class HelloModuleService {
-  getMessage() {
-    return "Hello, world!"
-  }
+```ts
+import { MedusaService } from "@medusajs/framework/utils"
+import Post from "./models/post"
+
+class BlogModuleService extends MedusaService({
+  Post,
+}){
 }
+
+export default BlogModuleService
 ```
 
-## 2. Export Module Definition
+## 3. Export Module Definition
 
 A module must have an `index.ts` file in its root directory that exports its definition. The definition specifies the main service of the module.
 
-For example, create the file `src/modules/hello/index.ts` with the following content:
+For example, create the file `src/modules/blog/index.ts` with the following content:
 
-```ts title="src/modules/hello.index.ts" highlights={[["4", "", "The main service of the module."]]}
-import HelloModuleService from "./service"
+```ts
+import BlogModuleService from "./service"
 import { Module } from "@medusajs/framework/utils"
 
-export const HELLO_MODULE = "hello"
+export const BLOG_MODULE = "blog"
 
-export default Module(HELLO_MODULE, {
-  service: HelloModuleService,
+export default Module(BLOG_MODULE, {
+  service: BlogModuleService,
 })
 ```
 
-## 3. Add Module to Configurations
-
-The last step is to add the module in Medusa’s configurations.
+## 4. Generate Migrations
 
-In `medusa-config.js`, add the module to the `modules` object:
+To generate migrations for your module, run the following command in the plugin's directory:
 
-```js title="medusa-config.js"
-import { HELLO_MODULE } from "./src/modules/hello"
-
-module.exports = defineConfig({
-  // ...
-  modules: [
-    {
-      resolve: "./modules/hello",
-    }
-  ]
-})
+```bash
+npx medusa plugin:db:genreate
 ```
 
 ## Use Module
 
-You can resolve the main service of the module in other resources, such as an API route:
+You can use the module in customizations within the plugin or within the Medusa application using this plugin. When the plugin is added to a Medusa application, all its modules are registered as well.
```

Comment:
Q: should we add a point about the fact that for now all plugins options are sent back to the module?

Reply:
Added 👍🏻